--- a/src/main/java/ggfab/mte/MTEAdvAssLine.java
+++ b/src/main/java/ggfab/mte/MTEAdvAssLine.java
@@ -205,6 +205,7 @@
     private int currentInputLength;
     private String lastStopReason = "";
     private int currentRecipeParallel = 1;
+    private int parallelForEnergy = 1;
 
     public MTEAdvAssLine(int aID, String aName, String aNameRegional) {
         super(aID, aName, aNameRegional);
@@ -388,6 +389,7 @@
         currentRecipe = null;
         currentInputLength = -1;
         currentRecipeParallel = 1;
+        parallelForEnergy = 1;
         stuck = false;
         baseEUt = 0;
         for (Slice slice : slices) {
@@ -641,7 +643,7 @@
             }
             if (slice.progress > 0) working++;
         }
-        lEUt = working * baseEUt;
+        lEUt = parallelForEnergy * working * baseEUt;
 
         if (lEUt > 0) {
             // overflow again :(
@@ -830,6 +832,7 @@
             }
 
             int currentParallelBeforeBatchMode = Math.min(currentParallel, maxParallelBeforeBatchMode);
+            parallelForEnergy = currentParallelBeforeBatchMode;
 
             calculator.setCurrentParallel(currentParallelBeforeBatchMode)
                 .calculate();
